src/lib.rs

```diff
@@ -8,163 +8,154 @@
-        metadata
-            .repository
-            .as_ref()
-            .cloned()
-            .unwrap_or_else(|| String::from("")),
-    );
-
-    let license = metadata
-        .license
-        .as_ref()
-        .cloned()
-        .unwrap_or_else(|| String::from("unknown license"));
+    let mut string = String::new();
+    File::open(&manifest)?.read_to_string(&mut string)?;
+
+    let parsed_manifest = string.parse::<Value>()?;
+    let table = &parsed_manifest
+        .as_table()
+        .ok_or_else(|| format_err!("Cargo manifest does not contain a toml table"))?;
+    let package = table
+        .get("package")
+        .ok_or_else(|| format_err!("Field \"package\" is missing in Cargo manifest"))?;
+    let name = read_from_manifest(package, &"name").unwrap_or_else(|| String::from(""));
+    let license = read_from_manifest(package, &"license").unwrap_or_else(|| String::from(""));
+    let description = read_from_manifest(package, &"description").unwrap_or_else(|| name.clone());
+    let homepage = read_from_manifest(package, &"homepage").unwrap_or_else(|| String::from(""));
+    let version = read_from_manifest(package, &"version").unwrap_or_else(|| String::from(""));
 
     // build up the ebuild path
-    let ebuild_path = PathBuf::from(format!("{}-{}.ebuild", package.name(), package.version()));
+    let path = PathBuf::from(ebuild_path.unwrap_or_else(|| format!("{}-{}.ebuild", name, version)));
+
+    let ebuild_path = if path.is_dir() {
+        if !path.exists() {
+            return Err(format_err!("No such file or directory"));
+        }
+        let ebuild_name = PathBuf::from(format!("{}-{}.ebuild", name, version));
+        path.join(ebuild_name)
+    } else {
+        path
+    };
 
     // Open the file where we'll write the ebuild
-    let mut file = try!(
-        OpenOptions::new()
-            .write(true)
-            .create(true)
-            .truncate(true)
-            .open(&ebuild_path)
-            .chain_err(|| "failed to create ebuild")
-    );
+    let mut file = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(&ebuild_path)?;
 
     // write the contents out
-    try!(
-        write!(
-            file,
-            include_str!("ebuild.template"),
-            description = desc.trim(),
-            homepage = homepage.trim(),
-            license = license.trim(),
-            crates = crates.join(""),
-            cargo_ebuild_ver = env!("CARGO_PKG_VERSION"),
-            this_year = 1900 + time::now().tm_year,
-        ).chain_err(|| "unable to write ebuild to disk")
-    );
+    writeln!(
+        file,
+        include_str!("ebuild.template"),
+        description = description.trim(),
+        homepage = homepage.trim(),
+        license = license.trim(),
+        crates = crates.join("\n"),
+        cargo_ebuild_ver = &env!("CARGO_PKG_VERSION"),
+        this_year = 1900 + time::now().tm_year
+    )?;
```

Comment:
ok this is a trivial code style change and I can break it out of this larger change.